### PR TITLE
refactor: update plugin for notation package

### PR DESCRIPTION
**Pull request by JeyJeyGao**
Replaced the old internal/plugin with new plugin package in notation package

Signed-off-by: Junjie Gao <junjiegao@microsoft.com>

### CODE REVIEW

**Review by shizhMSFT — APPROVED** [review]
LGTM

File: internal/mock/mocks.go [hunk 1]
```diff
@@ -101,40 +101,56 @@ func (t Repository) PushSignature(ctx context.Context, mediaType string, blob []
...
+
+func (p *PluginMock) GetMetadata(ctx context.Context, req *proto.GetMetadataRequest) (*proto.GetMetadataResponse, error) {
+	return &p.Metadata, nil
+}
+
+func (p *PluginMock) VerifySignature(ctx context.Context, req *proto.VerifySignatureRequest) (*proto.VerifySignatureResponse, error) {
+	if resp, ok := p.ExecuteResponse.(*proto.VerifySignatureResponse); ok {
+		return resp, nil
+	}
+	return nil, nil
```
Comment (patrickzheng200) [inline internal/mock/mocks.go:117]: It doesn't look correct to return nil for both of the outputs. How about:
`return nil, p.ExecuteError`. 
For details: please check https://github.com/patrickzheng200/notation-go/blob/refactor/internal/mock/mocks.go#L118

File: internal/mock/mocks.go [hunk 2]
```diff
@@ -101,40 +101,56 @@ func (t Repository) PushSignature(ctx context.Context, mediaType string, blob []
 	return ocispec.Descriptor{}, ocispec.Descriptor{}, nil
 }
 
+type PluginMock struct {
+	Metadata        proto.GetMetadataResponse
+	ExecuteResponse interface{}
```
Comment (patrickzheng200) [inline internal/mock/mocks.go:106]: Given the change below, this should be:
```suggestion
	ExecuteResponse interface{}
	ExecuteError    error
```

File: internal/mock/mocks.go [hunk 3]
```diff
@@ -101,40 +101,56 @@ func (t Repository) PushSignature(ctx context.Context, mediaType string, blob []
...
 			Name:                      "plugin-name",
 			Description:               "for mocking in unit tests",
 			Version:                   "1.0.0",
 			URL:                       ".",
 			SupportedContractVersions: []string{"1.0"},
 			Capabilities:              pm.PluginCapabilities,
 		},
-		Path: ".",
-		Err:  nil,
+		ExecuteResponse: pm.PluginRunnerExecuteResponse,
```
Comment (patrickzheng200) [inline internal/mock/mocks.go:150]: Given the changes above, this should be:
```suggestion
		ExecuteResponse: pm.PluginRunnerExecuteResponse,
		ExecuteError:    pm.PluginRunnerExecuteError,
```

**Review by patrickzheng200 — CHANGES_REQUESTED** [review]

File: internal/mock/mocks.go (hunk 1 above)
Comment (JeyJeyGao) [reply internal/mock/mocks.go]: Updated.

File: internal/mock/mocks.go (hunk 2 above)
Comment (JeyJeyGao) [reply internal/mock/mocks.go:106]: Updated.

File: internal/mock/mocks.go (hunk 3 above)
Comment (JeyJeyGao) [reply internal/mock/mocks.go:151]: Updated.

**Review by patrickzheng200 — APPROVED** [review]
LGTM

**Review by rgnote — APPROVED** [review]

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**